purchase_order_approval_block/models/purchase_order.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import _, api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    approval_block_id = fields.Many2one(
+        comodel_name="purchase.approval.block.reason", string="Approval Block Reason",
+    )
+    approval_blocked = fields.Boolean(
+        "Approval Blocked", compute="_compute_approval_blocked",
+    )
+
+    @api.depends("approval_block_id")
+    def _compute_approval_blocked(self):
+        for rec in self:
+            rec.approval_blocked = rec.approval_block_id
```

Comment:
Approval Blocked boolean field expects True or False
```suggestion
        for rec in self:
            if rec.approval_block_id:
                rec.approval_blocked = True
            else:
                rec.approval_blocked = False
```

Reply:
When assigning the value, it is transforming the new value to cache. So It is using the following function:
https://github.com/odoo/odoo/blob/13.0/odoo/fields.py#L1180

It is not needed.

Reply:
Ok!